File: dotnet-backend/projekatFlutter/Models/Alert.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class Alert:IEntity<int>
    {
        [Key]
        public int Id { get; set; }
        
        public int? VehicleId { get; set; }

        [MaxLength(450)]
        public string? DriverId { get; set; }

        public int? TripId { get; set; }

        public int? GeofenceId { get; set; }

        [Required]
        [MaxLength(50)]
        public string AlertType { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        public string Severity { get; set; } = "Info";

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [MaxLength(1000)]
        public string Message { get; set; } = string.Empty;

        [Column(TypeName = "decimal(10,8)")]
        public decimal? Latitude { get; set; }

        [Column(TypeName = "decimal(11,8)")]
        public decimal? Longitude { get; set; }

        public string? MetaData { get; set; }

        [Required]
        [MaxLength(30)]
        public string Status { get; set; } = "New";

        [MaxLength(450)]
        public string? AcknowledgedBy { get; set; }

        public DateTime? AcknowledgedAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        
        [ForeignKey("VehicleId")]
        public Vehicle? Vehicle { get; set; }

        [ForeignKey("DriverId")]
        public User? Driver { get; set; }

        [ForeignKey("TripId")]
        public Trip? Trip { get; set; }

        [ForeignKey("GeofenceId")]
        public Geofence? Geofence { get; set; }

        [ForeignKey("AcknowledgedBy")]
        public User? AcknowledgedByUser { get; set; }
    }
}

File: frontend/lib/models/alert.dart
class Alert {
  final int id;
  final int? vehicleId;              
  final String? driverId;            
  final int? tripId;                 
  final int? geofenceId;             
  final String alertType;
  final String severity;
  final String? title;              
  final String message;
  final double? latitude;          
  final double? longitude;           
  final String? metaData;            
  final String status;
  final String? acknowledgedBy;      
  final DateTime? acknowledgedAt;    
  final DateTime createdAt;
  final DateTime? updatedAt;         

  Alert({
    required this.id,
    this.vehicleId,                  
    this.driverId,
    this.tripId,
    this.geofenceId,
    required this.alertType,
    required this.severity,
    this.title,
    required this.message,
    this.latitude,
    this.longitude,
    this.metaData,
    required this.status,
    this.acknowledgedBy,
    this.acknowledgedAt,
    required this.createdAt,
    this.updatedAt,
  });

  factory Alert.fromJson(Map<String, dynamic> json) {
    return Alert(
      id: json['id'] as int,
      vehicleId: json['vehicleId'] as int?,
      driverId: json['driverId'] as String?,
      tripId: json['tripId'] as int?,
      geofenceId: json['geofenceId'] as int?,
      alertType: json['alertType'] as String,
      severity: json['severity'] as String,
      title: json['title'] as String?,
      message: json['message'] as String,
      latitude: json['latitude'] != null ? (json['latitude'] as num).toDouble() : null,
      longitude: json['longitude'] != null ? (json['longitude'] as num).toDouble() : null,
      metaData: json['metaData'] as String?,
      status: json['status'] as String,
      acknowledgedBy: json['acknowledgedBy'] as String?,
      acknowledgedAt: json['acknowledgedAt'] != null 
          ? DateTime.parse(json['acknowledgedAt'] as String)
          : null,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vehicleId': vehicleId,
      'driverId': driverId,
      'tripId': tripId,
      'geofenceId': geofenceId,
      'alertType': alertType,
      'severity': severity,
      'title': title,
      'message': message,
      'latitude': latitude,
      'longitude': longitude,
      'metaData': metaData,
      'status': status,
      'acknowledgedBy': acknowledgedBy,
      'acknowledgedAt': acknowledgedAt?.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}